Source/bmalloc/libpas/src/libpas/pas_local_view_cache.h

```diff
@@ -94,8 +94,31 @@ static inline bool pas_local_view_cache_is_empty(pas_local_view_cache* cache)
 
 static inline bool pas_local_view_cache_prepare_to_pop(pas_local_view_cache* cache)
 {
+    /* TLC decommit can decommit pas_local_view_cache at any time if it is stopped state. Thus in this code, we need to assume that
+       if it is stopped state, it can be cleared to decommitted state at any time.
+       We would like to ensure that this pas_local_view_cache is committed after this function.
+       To ensure that, we first need to check cache->scavenger_data.kind
+       to see that this is pas_local_allocator_view_cache_kind. Since decommit happens after setting it to pas_local_allocator_decommitted_kind,
+       we can ensure that (1) is_in_use is true, (2) this is not a stopped state, and (3) this will not be decommitted.
+       But if it is not, then this local_view_cache is stopped or decommitted and stopped state can be changed to decommitted state at any time.
+       In that case, we need to revive it to active state. */
     pas_local_view_cache_state state;
-    
+
+    if (PAS_UNLIKELY(cache->scavenger_data.kind != pas_local_allocator_view_cache_kind)) {
+        /* It is stopped (it can be decommitted at any time now) or decommitted.
+           But we found that this local_view_cache will be used now. So, this commits the view to revive it for the next use.
+           And in either case (stopped or decommitted), it is already empty. So we cannot pop anything from this local_view_cache,
+           returning false. */
+        pas_local_allocator_scavenger_data_commit_if_necessary_slow(
+            &cache->scavenger_data,
+            pas_local_allocator_scavenger_data_commit_if_necessary_slow_is_in_use_with_no_locks_held_mode,
+            pas_local_allocator_view_cache_kind);
+        return false;
+    }
+
+    /* Because kind is pas_local_allocator_view_cache_kind and is_in_use is already true,
+       this cache will not be decommitted. Safely use it */
+
```

Comment:
In fact, it can be `false` (super fun part). Yeah, we set it to `true`. But it is possible that this local_view_cache's page is madvise-decommitted just after that. And OS could already collected this backing page. In that case, when accessing this field, it is zero page, so we will get `false`!
The invariant is that, after returning this function, `view_cache->scavenger_data.is_in_use` must be `true`. In `pas_local_allocator_scavenger_data_commit_if_necessary_slow`, we set it to `true` during reviving. I'll add this to the caller of this function.